dev/NumberBox/NumberBoxParser.h

```diff
@@ -0,0 +1,78 @@
+﻿#pragma once
+#include <vector>
+#include <string>
+#include <sstream>
+#include <iostream>
+#include <cmath>
+#include <algorithm>
+#include <iterator>
+#include <cstdlib>
+#include <regex>
+#include <stack>
+#include <optional>
+
+// Represents a token in a mathematic expression. This can be a simple operator (+-*/^), a decimal number, or parentheses. 
+class MathToken
+{
+public:
+    enum TokenType
+    {
+        Invalid = 0,
+        Numeric = 1,
+        Operator = 2,
+        ParenLeft = 3,
+        ParenRight = 4,
+        EOFToken = 6
+    };
+    TokenType type;
+    std::wstring str;
+    MathToken();
+    MathToken(TokenType t, std::wstring s);
+    MathToken(TokenType t, std::wstring& s);
+
+};
+
+// Handles tokenizing strings
+class MathTokenizer
+{
+    private:
```

Comment:
The convention in our codebase is that public members are first, private are last. Fields should be at the very bottom of the class and grouped together.